src/screens/AddSiteScreen.js

```diff
@@ -1,204 +1,339 @@
+    };
+
+    onUpdateLongitudeText = (newLongText) => {
+        this.props.updateLongitudeText({longitudeText: newLongText})
+    };
+
+    onUpdateSiteTitleText = (newTitleText) => {
+        this.props.updateSiteTitleText({siteTitleText: newTitleText})
+    };
+
+    onUpdateSiteDescriptionText = (newDescriptionText) => {
+        this.props.updateSiteDescriptionText({siteDescriptionText: newDescriptionText})
+    };
+
+    onUpdateSiteDirectionsText = (newDirectionsText) => {
+        this.props.updateSiteDirectionsText({siteDirectionsText: newDirectionsText})
+    };
+
+    onUpdateSiteNearestTownText = (newNearestTownText) => {
+        this.props.updateSiteNearestTownText({siteNearestTownText: newNearestTownText})
+    };
+
+    onUpdateAccessibilityOption = (newAccessibilityOption) => {
+        this.props.updateAccessibilityOption({accessibilityOption: newAccessibilityOption})
+    };
+
+    onUpdateFacilitiesOption = (newFacilitiesOption) => {
+        this.props.updateFacilitiesOption({facilitiesOption: newFacilitiesOption})
+    };
+
+    onClickReset = () => {
+        this.props.resetAddScreenFields();
+    };
+
+    render() {
+        const {buttonStyle, headerTitle, largeTextInput, modalStyle, lastButtonStyle, exitOrResetStyle} = styles;
+        const {latitudeText, longitudeText, siteTitleText, siteDescriptionText, siteDirectionsText, siteNearestTownText, accessibilityOption, facilitiesOption} = this.props;
+
+        return (
+            <View>
+                <Card>
+                    <Text>
+                        {site_description}
+                    </Text>
+                </Card>
+                <Button
+                    onPress={() => this.openModal()}
+                    title="Open modal"
+                    large
+                    rounded={true}
+                    buttonStyle={buttonStyle}
+                    icon={{name: 'plus', type: 'font-awesome'}}
+                    title={add_site}
+                >
+                    {upload}
+                </Button>
+
+                <Modal
+                    visible={this.state.modalVisible}
+                    animationType={'slide'}
+                    onRequestClose={() => this.closeModal()}
+                >
+                    <ScrollView style={modalStyle}>
+                        <View style={exitOrResetStyle}>
+                            <Icon
+                                type='font-awesome'
+                                name='times-circle'
+                                onPress={() => this.closeModal()}
+                            />
+
+                            <Button
+                                title="Reset"
+                                onPress={this.onClickReset}
+                                backgroundColor="rgba(0,0,0,0)"
+                                color="rgba(0,122,255,1)"
+                            />
+
+                        </View>
```

Comment:
@amgoodman5 I put the 'close' icon and the 'reset' button in a view that has `exitOrResetStyle ` to space them out across the top of the modal. PLEASE feel free to style this some other way to make them look better.

Reply:
Yes of course